### ISSUE TITLE
[expo-image-picker] When using launchImageLibraryAsync, modal snaps to the top of the screen

### ISSUE THREAD

**Issue by cwnicoletti** [expo/expo#15214]
### Summary

This bug only happens when you set `allowsEditing: true` for `ImagePicker.launchImageLibraryAsync`. The modal does its usual animation transitioning from the bottom to the top of the screen, and at the end of the animation, it snaps to the top. This puts the modal underneath the front-facing built-in microphone and camera on the **iPhone 12**.

When `allowsEditing: false`, this doesn't happen at all.

Also wanted to note that there's a work around for this. If you set `"UIViewControllerBasedStatusBarAppearance": "YES"` in `"infoPlist"`, this will fix the issue, but only when you `expo build:ios` and simulate/archive.

```
{
  "expo": {
    "name": "ExampleProject",
    "slug": "ExampleProject",
    "version": "0.0.1",
    "orientation": "portrait",
    "ios": {
      "infoPlist": {
        "UIViewControllerBasedStatusBarAppearance": "YES" // <-- here
      }
    },
}
```

iOS version: 14.8.1

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

43

### Environment

Expo CLI 4.13.0 environment info:
    System:
      OS: Windows 10 10.0.19043
    Binaries:
      Node: 14.17.3 - C:\Program Files\nodejs\node.EXE
      npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  4.0.0.0 AI-193.6911.18.40.6626763
    npmPackages:
      expo: ~43.0.2 => 43.0.3
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
      react-native-web: 0.17.1 => 0.17.1
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

Starting from the blank project:

```
export default function App() {
  const imagePickerFunc = () => {
    ImagePicker.launchImageLibraryAsync({
      allowsEditing: true,
    });
  };

  return (
    <View style={styles.container}>
      <Button
        title="Press me"
        onPress={() => {
          imagePickerFunc();
        }}
      ></Button>
      <StatusBar style="auto" />
    </View>
  );
}
```

**Comment by sanborN93** [comment]
Hey,
i had the same issue and used this as a workaround.

```
 const pickerResult = await ImagePicker.launchImageLibraryAsync({
                mediaTypes: ImagePicker.MediaTypeOptions.Images,
                presentationStyle: UIImagePickerPresentationStyle.OverCurrentContext,
                allowsEditing: false,
                quality: 1,
            });
```

By adding a presentationStyle u can change the behavior of the modal.

**Comment by cwnicoletti** [comment]
Hey @sanborN93,
So I originally tried adding `presentationStyle: ImagePicker.UIImagePickerPresentationStyle.OverCurrentContext` like you said and I got an error:
`[Unhandled promise rejection: TypeError: undefined is not an object (evaluating 'ImagePicker.UIImagePickerPresentationStyle.OverCurrentContext')]`

and then I tried exactly what you said `presentationStyle: UIImagePickerPresentationStyle.OverCurrentContext` and got the error:
`[Unhandled promise rejection: ReferenceError: Can't find variable: UIImagePickerPresentationStyle]
`

So then I referred to the area in the doc you linked which assigned numeric values to each presentation style and attempted:
`presentationStyle: 6`
However that didn't work either. Maybe I'm missing something here.

**Comment by sanborN93** [comment]
Hey @cwnicoletti,

i guess you simply didn't import ImagePicker.UIImagePickerPresentationStyle.OverCurrentContext.
I am not at my workplace right now, so i can't look it up.

But i am sure you have to import it separately.

**Comment by sanborN93** [comment]
i guess its something like
`import{UIImagePickerPresentationStyle.OverCurrentContext} from expo-image-picker `

or smth

**Comment by cwnicoletti** [comment]
oh yeah don't worry about it.

So I tried importing that: `import { UIImagePickerPresentationStyle } from 'expo-image-picker/build/ImagePicker.types';`
and calling launchImageLibraryAsync as: 
```
const result = await ImagePicker.launchImageLibraryAsync({
      presentationStyle: UIImagePickerPresentationStyle.OverCurrentContext,
    });
```
and now I'm getting: 
`[Unhandled promise rejection: TypeError: undefined is not an object (evaluating '_ImagePicker.UIImagePickerPresentationStyle.OverCurrentContext')]`

So I tried just: `import { UIImagePickerPresentationStyle } from "expo-image-picker";`
yet still I'm getting:
`[Unhandled promise rejection: TypeError: undefined is not an object (evaluating 'ImagePicker.UIImagePickerPresentationStyle.OverCurrentContext')]`

**Comment by sanborN93** [comment]
add these to your package.json

"expo-image-picker": "~11.0.3",
"expo-image-picker/build/ImagePicker.types": "~11.0.3"

and run npm install or yarn

if you dont use SDK43 you have to use version according to your sdk

**Comment by sanborN93** [comment]
I made a quick snack here: 
https://snack.expo.dev/@lenb/expo-image-picker-snack

**Comment by cwnicoletti** [comment]
Awesome. That works now, I was just using SDK42 and updating it to 43 got that `presentationStyle` and `UIImagePickerPresentationStyle.OverCurrentContext` to work. Thanks for that

However, it seems that it's still snapping with certain presentation styles, and the rest of the presentation styles fill the entire screen. Which is different from the desired results, which `"UIViewControllerBasedStatusBarAppearance": "YES"` gives me

[Demo of desired result (UIViewControllerBasedStatusBarAppearance workaround)](https://res.cloudinary.com/personaluse1234/video/upload/v1638408905/Video_tw8klg.mov)
[Demo of snapping](https://res.cloudinary.com/personaluse1234/video/upload/v1638408905/Video_2_awp2wa.mov)
[Demo of fullscreen](https://res.cloudinary.com/personaluse1234/video/upload/v1638408904/Video_1_mf7hls.mov)

So it seems, as far as the presentation styles, I'm getting:

UIImagePickerPresentationStyle.FullScreen -> fullscreen
UIImagePickerPresentationStyle.PageSheet -> snapping
UIImagePickerPresentationStyle.FormSheet -> snapping
UIImagePickerPresentationStyle.CurrentContext -> fullscreen
UIImagePickerPresentationStyle.OverFullScreen -> fullscreen
UIImagePickerPresentationStyle.OverCurrentContext -> fullscreen
UIImagePickerPresentationStyle.Popover -> snapping
UIImagePickerPresentationStyle.Automatic -> snapping

**Comment by sanborN93** [comment]
@cwnicoletti you're welcome.

However, my workaround does not fundamentally solve the problem. If you want to keep the on top behavior/sheet style, you currently have to live with the bug. Unfortunately, there are problems with the album selection on iPhones, which got the notch. The snapping bug tends to leave the album selection behind the notch, which makes it hard to reach it.
The only workaround is to use the Image Picker full-screen, as u mentioned with one of the styles above. 
So basically we have to wait for a complete fix.
